vyper/ast/folding.py

```diff
@@ -111,6 +120,7 @@ def replace_builtin_functions(vyper_module: vy_ast.Module) -> int:
             continue
         try:
             new_node = func.evaluate(node)  # type: ignore
+            new_node._metadata["type"] = node._metadata.get("type")
```

Comment:
i don't think we want to assign anything here unless the result of `get()` is not None

Reply:
if we know `"type"` should be in the `_metadata` dict, use `["type"]`